### PR TITLE
Miscellaneous Updates

### PR DESCRIPTION
**Pull request by mchurchf**
Not great code development practice, but here are a lot of small updates to the code.
-Ability to do perfect restarts.
-Fix to actuator line to deal with the fact that Aerodyn and Elastodyn points are not perfectly collocated
-Various small reorganizations.
-Perturbation code updates.

### CODE REVIEW

File: Allwmake [hunk 1]
```diff
@@ -71,7 +71,7 @@ else
     if [ -d ${OPENFAST_DIR} ]
     then
         echo "OPENFAST_DIR is set to '${OPENFAST_DIR}'. Attempting to compile OpenFAST-coupled solver."
-        wmake applications/solvers/superDeliciousVanilla.turbine
+       #wmake applications/solvers/superDeliciousVanilla.turbine
```
Comment (ewquon) [inline Allwmake:74]: @mchurchf did you mean to leave this commented out?

File: applications/solvers/superDeliciousVanilla/TEqn.H [hunk 2]
```diff
@@ -9,11 +9,11 @@
...
       - fvm::laplacian(kappaEff, T)
-      - fvc::div(qwall)
-      - temperatureGeoMesoTerm.force()
-      - temperaturePerturbationZones.source()
      ==
-        fvOptions(T)
+        fvc::div(qwall)
+      + temperatureGeoMesoTerm.force()
+      + temperaturePerturbationZones.source()
+      + fvOptions(T)
```
Comment (ewquon) [inline applications/solvers/superDeliciousVanilla/TEqn.H:16]: Sign change here, which is correct? Do we even use fvOptions?

File: applications/solvers/superDeliciousVanilla/UEqn.H [hunk 3]
```diff
@@ -3,15 +3,17 @@
...
-     ==
-        fvOptions(U)
+    ==
+      - fvc::div(Rwall)
+      + Coriolis.force()
+      + momentumGeoMesoTerm.force()
+      + momentumSpongeLayers.force()
+      + momentumPerturbationZones.source()
+      - (1.0 - scalar(bodyForceStabilization)) * actuatorTurbineArray.source()
+      + fvOptions(U)
```
Comment (ewquon) [inline applications/solvers/superDeliciousVanilla/UEqn.H:16]: Ditto about the sign on  `fvOptions(U)`

File: applications/solvers/superDeliciousVanilla/pEqn.H [hunk 4]
```diff
@@ -26,9 +28,9 @@
     // applies the correction indirectly by setting the pressure gradient to be used in
     // the fixedFluxPressure boundary condition on p_rgh, or directly if the zeroGradient
     // boundary condition on p_rgh is used.
-    surfaceScalarField phiFixedFlux = phiHbyA;
+    surfaceScalarField phiFixedFlux = 1.0*phiHbyA;
```
Comment (ewquon) [inline applications/solvers/superDeliciousVanilla/pEqn.H:31]: What does this 1.0 factor do?

**Review by ewquon — COMMENTED** [review]
Pretty substantial overhaul to enable a lot of the recent progress, looks good to me, just a few minor questions.

File: Allwmake (hunk 1 above)
Comment (rthedin) [reply Allwmake:74]: The `.turbine` solver is now obsolete. The regular `superDeliciousVanilla` handles turbine and no-turbine cases. Caveat: needs to be compiled with OpenFAST coupling even if you don't plan to use turbines.

Comment (rthedin) [reply Allwmake]: Fixed by [b7b82cd](https://github.com/NREL/SOWFA-6/pull/50/commits/b7b82cd4a958f14a3f4e6b5c9a692b4757bd901b)

**Comment by rthedin** [comment]
@mchurchf Should we entirely remove the `.turbine` version of the solver?

**Review by rthedin — APPROVED** [review]

**Pull request merged**